backend/danswer/search/models.py

```diff
@@ -32,6 +34,37 @@ class BaseFilters(BaseModel):
     time_cutoff: datetime | None = None
     tags: list[Tag] | None = None
 
+    @classmethod
+    def from_persona(cls, persona: Persona) -> "BaseFilters | None":
+        if persona.use_recent_documents and persona.num_days is not None:
+            from_date = datetime.now(timezone.utc) - timedelta(days=persona.num_days)
+            return cls(time_cutoff=from_date)
+        else:
+            return None
+
+    # Merges two filters into one, giving preference to the human-specified retrieval filters
+    @classmethod
+    def merge(
+        cls,
+        retrieval_filter: "BaseFilters | None",
+        persona_filter: "BaseFilters | None",
+    ) -> "BaseFilters | None":
+        if retrieval_filter is None and persona_filter is None:
+            return None
+        elif retrieval_filter is None:
+            return persona_filter
+        elif persona_filter is None:
+            return retrieval_filter
```

Comment:
doesn't this run the risk of overwriting other filters passed in?

Reply:
yes, it may have a pass by reference/value issue.  Will search for the right way to address this.